cmd/remote-session.go

```diff
@@ -0,0 +1,222 @@
+// `podman --remote exec`.
+func runExec(c *cobra.Command, args []string) error {
+	if err := checkEnv(); err != nil {
+		return err
+	}
+	session := os.Getenv("COREOS_ASSEMBLER_REMOTE_SESSION")
+	podmanargs := []string{"--remote", "exec", "-it", session, "cosa"}
+	podmanargs = append(podmanargs, args...)
+	cmd := exec.Command("podman", podmanargs...)
+	cmd.Stdin = os.Stdin
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	if err := cmd.Run(); err != nil {
+		if exitError, ok := err.(*exec.ExitError); ok {
+			// If the command failed let's exit with the same exitcode
+			// as the remotely executed process.
+			os.Exit(exitError.ExitCode())
+		} else {
+			return err
+		}
+	}
+	return nil
+}
+
+// Executes a `podman --remote ps -a --filter id=<container>`
+// to show the status of the remote running cosa container.
+func runPS(c *cobra.Command, args []string) error {
+	if err := checkEnv(); err != nil {
+		return err
+	}
+	session := os.Getenv("COREOS_ASSEMBLER_REMOTE_SESSION")
+	podmanargs := []string{"--remote", "ps", "-a",
+		fmt.Sprintf("--filter=id=%s", session)}
+	cmd := exec.Command("podman", podmanargs...)
+	cmd.Stdin = os.Stdin
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	return cmd.Run()
+}
+
+// runSync provides an rsync-like interface that allows
+// files to be copied to/from the remote. It uses
+// `podman --remote exec` as the transport for rsync (see [1])
+//
+// One of the arguments here must be prepended with a `:`. This
+// argument will represent the path on the remote. This function
+// will substitute `:` with `$COREOS_ASSEMBLER_REMOTE_SESSION:`.
+// That environment var just contains the container ID on the remote.
+//
+// [1] https://github.com/moby/moby/issues/13660
+func runSync(c *cobra.Command, args []string) error {
+	// check environment
+	if err := checkEnv(); err != nil {
+		return err
+	}
+	// check arguments. Need one with pre-pended ':'
+	found := 0
+	for index, arg := range args {
+		if strings.HasPrefix(arg, ":") {
+			args[index] = fmt.Sprintf("%s%s",
+				os.Getenv("COREOS_ASSEMBLER_REMOTE_SESSION"), arg)
+			found++
+		}
+	}
+	if found != 1 {
+		return fmt.Errorf("Must pass in a single arg to rsync with `:` prepended")
+	}
+	// build command and execute
+	rsyncargs := []string{"-avh", "--mkpath", "--blocking-io",
+		"--rsh", "podman --remote exec --user root -i"}
+	rsyncargs = append(rsyncargs, args...)
+	cmd := exec.Command("rsync", rsyncargs...)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	return cmd.Run()
+}
+
+func init() {
+	cmdRemoteSession.AddCommand(cmdRemoteSessionCreate)
+	cmdRemoteSession.AddCommand(cmdRemoteSessionDestroy)
+	cmdRemoteSession.AddCommand(cmdRemoteSessionExec)
+	cmdRemoteSession.AddCommand(cmdRemoteSessionPS)
+	cmdRemoteSession.AddCommand(cmdRemoteSessionSync)
+
+	// cmdRemoteSessionCreate options
+	cmdRemoteSessionCreate.Flags().StringVarP(
+		&remoteSessionOpts.Image, "image", "",
+		"quay.io/coreos/coreos-assembler:latest",
```

Comment:
Shouldn't this be `quay.io/coreos-assembler/coreos-assembler:latest`? My reading of https://github.com/coreos/fedora-coreos-tracker/issues/1171 is that we decided to leave it there for now.

Reply:
yep. I just messed up when I wrote it. Will be fixed soon.